### DISCUSSION TITLE
Download/Export as PDF

### DISCUSSION THREAD

**Discussion by tvarsis** [outline/outline#2461] [Ideas]
Right now it is only possible to download a doc as md format, which is kind of a "developer" format. The average user would expect to be able to download a doc as PDF instead to use offline, which I think would be the expected result outside the dev perspective.

**Comment by tommoor** [comment]
Hey @tvarsis – while I agree this would be nice to have it is actually very complicated to do well. However, all browsers now enable you to "Print to PDF", so you can use that option which will correctly use the print stylesheet.

![image](https://user-images.githubusercontent.com/380914/130024708-61c8ed60-ed68-490c-bb32-884bb205c64c.png)

**Reply by Nicicalu** [reply]
@tommoor Two things I've noticed using "Save as PDF"

1. Block Quotes aren't formatted correclty (the thin grey bar on the left of the quote is not visible)
2. Code is formatted correctly but there is a scrollbar if it overflows horizontally. In a PDF you can't click scrollbars which means you can't read the code that overflows. Is that on purpose?

My info:

**Outline:**
- Install: Self hosted
- Version: v.0.65.2 

**Desktop:**
 - OS: Windows 10
 - Browser: Google Chrome
 - Version: 103.0.5060.114

**Reply by tommoor** [reply]
> Is that on purpose?

I think it just hasn't been considered yet. I suppose the only option is to wrap code when printing

**Reply by Nicicalu** [reply]
> > Is that on purpose?
> 
> I think it just hasn't been considered yet. I suppose the only option is to wrap code when printing

What do you mean with "wrap the code"? Something in the code or something the user has to do?

**Reply by tommoor** [reply]
change the display to wrap long lines, it would be a print-only css change

**Reply by Nicicalu** [reply]
@tommoor How about the thing I mentioned about the quote not being rendered right?

**Reply by tommoor** [reply]
Feel free to file an issue that covers both, they are both bugs 😄

**Reply by Nicicalu** [reply]
> Feel free to file an issue that covers both, they are both bugs 😄

Oh, yeah sure. Thanks

**Comment by tvarsis** [comment]
@tommoor Thanks for your reply! Yeah, implementing a PDF exporter from scratch for sure is a lot of work. But there are tons of ready made npm modules that handle that process pretty straight forward, which would lead to a more seamless experience than using the native print to pdf option. Many users doesn't even know about that option and it can in it self give very sub-optimal results at times. Maybe testing out some of the ready made npm modules as a POC to see how it works could at least be an option for consideration?

**Reply by tommoor** [reply]
Yep, I was just giving an option for "in the meantime" – using a module for this is a must, but it still requires us to be able to accurately render documents on the server. This is something that's in progress for adding diffs to notification emails so it's part way there – I think the work can be used for PDF export once we have an accurate HTML export with styles.

**Reply by tvarsis** [reply]
@tommoor Thanks for your reply! That make sense, thanks!

**Reply by roovyshapiro** [reply]
> adding diffs to notification emails

This is an amazing idea!

**Reply by almereyda** [reply]
The diffs for email notifications are discussed in #1146

**Comment by Nicicalu** [comment]
Exports to PDF would be very nice!

**Comment by edelbende** [comment]
Very nice and needed, I confirm !
In my ideal representation, one could select the nested documents to be exported, with the possibility of automatically generating a summary with page numbers...the dream ! 😉

**Comment by Ivan989898** [comment]
Please, enable export PDF in Community Edition: https://www.getoutline.com/changelog/pdf-html-export

**Reply by tommoor** [reply]
Hi, I want to be clear that PDF export won't be coming to the community edition – but you can always use the browsers print functionality to print to PDF.

**Reply by Altagen** [reply]
Hey !

I would like to know what is considered a "Community edition"?

Is self managed considered a CE ? If so, it's not clearly written in the plan comparison.

Right now there is only :

SAML 2.0 (Okta, OneLogin), Confluence importer, Audit trail, Setup and maintenance support, Analytics and reporting

That are different from the paid plan.

I also wanted to point that the html that is exported does not support basic language (like accentuated characters : é, è (maybe utf-32 or rfc should be used)) 
And it only exports the strings, not the image you could have put.

It could be interesting to put all the images in a folder and replace by <omg src="foldername/img">

And zip the content. Doing so, you can have a working little website, and you don't need to modify yourself the html once it's generate, you only need to click on the html. (also it could be great to also seperate the css but i understand the need of having a simple html file.


Finally, if self-managed isn't considered CE, how is it possible to have the PDF unlocked in self managed ?

Thanks,
Ange

**Reply by tommoor** [reply]
PDF generation is a separate service that we run for the cloud-hosted edition. Due to the way that PDF generation works it's not really feasible to bundle it with the self managed images (either Community or Enterprise), this is why it's only available on Cloud.

You can always print to PDF though which is really not very different.

**Reply by Altagen** [reply]
Yeah not really... Delivering the print page with the lack of good layout can be an issue. But as it's a cloud service, I understand that there won't be any advancement in that sens.

Maybe if it is possible to do the adjustments for the HTML, we could fix the problem by using any good html to pdf converter.
